pkg/cmd/pulumi/convert.go

```diff
@@ -97,23 +97,17 @@ func newConvertCmd() *cobra.Command {
 
 // pclGenerateProject writes out a pcl.Program directly as .pp files
 func pclGenerateProject(directory string, project workspace.Project, p *pcl.Program) error {
-	if directory != "." {
-		err := os.MkdirAll(directory, 0o755)
+	if directory == "." {
+		cwd, err := os.Getwd()
 		if err != nil {
-			return fmt.Errorf("could not create output directory: %w", err)
+			return err
 		}
+		directory = cwd
 	}
 
 	// We don't write out the Pulumi.yaml for PCL, just the .pp files.
-	for file, source := range p.Source() {
-		outputFile := path.Join(directory, file)
-		err := os.WriteFile(outputFile, []byte(source), 0o600)
-		if err != nil {
-			return fmt.Errorf("could not write output program: %w", err)
-		}
-	}
-
-	return nil
+	fs := afero.NewOsFs()
+	return p.WriteSourceFiles(directory, fs)
```

Comment:
Given you can do `afero.NewBasePathFs(afero.NewOsFs(), directory)` with afero, WriteSourceFiles could just take the FS alone, no need for a directory path as well.

Reply:
Yeah I could do that, but I need to pass a directory anyways when writing source files of components so it makes sense that we start with the root directory. It is basically the same thing

Reply:
Yeh we need to do that _internally_ in `writeSourceFiles` but the user doesn't need to care about that for `WriteSourceFiles`. Most of our other afero uses seem to be along this "just take an Fs and it should have the correct root" approach vs "take an Fs and a directory path in that fs" so I think it'd be slightly more consistent for `WriteSourceFiles` to just take the Fs.

Reply:
Done ✅